### PR TITLE
Add the codecov configuration

### PR DESCRIPTION
**Pull request by Aisuko**
Signed-off-by: aisuko <urakiny@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Aisuko** [comment]
Screenshot from 2020-11-16 19-05-19

**Comment by Aisuko** [comment]
Screenshot from 2020-11-16 19-08-21

**Comment by Aisuko** [comment]
Please kindly change the repo settings,  the PR need to be review finished before it be merged.

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        target: 50%
+        flags:
+          - api
+
+flags:
+  # filer the folder(s) you wish to measure by that flag
+  api:
+    # only inlcude files in the pkg
+    paths:
+      - pkg/
```
Comment (mgfeller) [inline codecov.yml:50]: can this be omitted to include all paths, or do we want to list all paths explicitly?

**Comment by mgfeller** [comment]
This looks very useful. I'm new to codecov, is this the currently most popular integrated code coverage solution?
Do we use that in other repositories already? As far as I understand, it cannot be run locally. Does it get triggered on push in forks?

**Review by mgfeller — CHANGES_REQUESTED** [review]

**Comment by Aisuko** [comment]
> This looks very useful. I'm new to codecov, is this the currently most popular integrated code coverage solution?
> Do we use that in other repositories already? As far as I understand, it cannot be run locally. Does it get triggered on push in forks?

I remember I saw the codecov in kubernetes repo at the first but they removed codecov several weeks almost one year ago. I can confirm it was a popular solution one year ago but not sure in currently.

In fact, I do not have the repo permission, so I can not add the environment values into our repo, so just one this repo we plan to use, but I'd like to listen from you guys.

Yep, it add the comment to every PR which want to merge to master branch and can not run locally. In local, maybe we can add script in Makefile with other plugins.

File: codecov.yml (hunk 1 above)
Comment (Aisuko) [reply codecov.yml:50]: Nice catch, I want to list the path which includes more tool feature functions. What's your idea?

Comment (Aisuko) [reply codecov.yml]: The configuration `api` is target 50%, and the rest of file may cover by default.

**Comment by mgfeller** [comment]
Btw I fixed the failing golangci-lint in the today's commit in #17 .

**Review by mgfeller — APPROVED** [review]

File: codecov.yml (hunk 1 above)
Comment (mgfeller) [reply codecov.yml]: I think we could start using it to gain experience and adjust the configuration after that.

It makes sense to have different configuration for different areas of the code, depending for instance on the risk.

As I don't yet know this particular tool, I have to get familiar with it first.

We have not tests yet, so there won't be much coverage....

**Comment by mgfeller** [comment]
> 
> I remember I saw the codecov in kubernetes repo at the first but they removed codecov several weeks almost one year ago. I can confirm it was a popular solution one year ago but not sure in currently.

Thanks for the info.

> 
> In fact, I do not have the repo permission, so I can not add the environment values into our repo, so just one this repo we plan to use, but I'd like to listen from you guys.

Not quite sure I understand this about environment variables..

> Yep, it add the comment to every PR which want to merge to master branch and can not run locally. In local, maybe we can add script in Makefile with other plugins.

Would be nice to have it on push to any branch, so that it also triggers in forks. I like to get feedback as early as possible.

**Comment by mgfeller** [comment]
We should start writing tests :-)

**Review by mgfeller — APPROVED** [review]
I would just start using it and gain experience with it.

File: codecov.yml (hunk 1 above)
Comment (Aisuko) [reply codecov.yml]: Agree with you, as the repo be used to other adapters we should set the restict rules for this repo's code, and as you said above we could start use it to gain experience.

**Comment by Aisuko** [comment]
The Github Action can pull  the environments what we set up to the repo settings

**Comment by kumarabd** [comment]
@Aisuko is this good for merging?

**Comment by Aisuko** [comment]
> @Aisuko is this good for merging?

Yep, We can merge this first.

**Review by kumarabd — APPROVED** [review]

**Pull request merged**